src/ident/AclIdent.cc

```diff
@@ -56,10 +54,8 @@ int
 ACLIdent::match(ACLChecklist *cl)
 {
     const auto checklist = Filled(cl);
-    if (checklist->rfc931[0]) {
-        return data->match(checklist->rfc931);
-    } else if (checklist->conn() != nullptr && checklist->conn()->clientConnection != nullptr && checklist->conn()->clientConnection->rfc931[0]) {
-        return data->match(checklist->conn()->clientConnection->rfc931);
+    if (checklist->ident()) {
+        return data->match(checklist->ident()->c_str());
```

Comment:
If, after addressing another change request, Ident::User starts using an empty value to represent errors, then this code needs to be adjusted to feed dash_str (instead of an empty value) to `data->match()`. However, this outer `if` statement should still check for ident() value existence to prevent repeated goAsync() calls below.

P.S. When you adjust this code, please do not call ident() twice in this `if` statement.

P.S. If official code feeds `data->match()` with dash_str (as I believe it does), then please also adjust `ident_regex` documentation in cf.data.pre to mention the special meaning of a dash value.